JSONObject.java

```diff
@@ -1759,12 +1763,22 @@ public boolean similar(Object other) {
             return false;
         }
     }
-
+    
+    /**
+     * Tests if the value should be tried as a decimal. It makes no test if there are actual digits.
+     * 
+     * @param val value to test
+     * @return true if the string is "-0" or if it contains '.', 'e', or 'E', false otherwise.
+     */
+    protected static boolean isDecimalNotation(final String val) {
+        return val.indexOf('.') > -1 || val.indexOf('e') > -1
+                || val.indexOf('E') > -1 || "-0".equals(val);
+    }
+    
```

Comment:
might be worth seeing how `new BigDecimal` and `parseNumber` check what's the decimal separator, the idea would be to have your function to be consistent with what Java does already, but again, maybe low priority, maybe not worth the effort.

If you run JVM in a French server for example, `Double.parseDouble` could expect a comma instead of dot, but I'm not sure of it.